lib/project_types/script/forms/ask_script_uuid.rb

```diff
@@ -6,9 +6,13 @@ class AskScriptUuid < ShopifyCLI::Form
       attr_reader :uuid
       def ask
         scripts = @xargs
+        if scripts.empty?
+          raise ShopifyCLI::Abort, ctx.message("script.error.no_scripts_found_in_app")
+        end
 
-        return if scripts.empty? ||
-          !CLI::UI::Prompt.confirm(ctx.message("script.application.ensure_env.ask_connect_to_existing_script"))
+        unless CLI::UI::Prompt.confirm(ctx.message("script.application.ensure_env.ask_connect_to_existing_script"))
+          raise ShopifyCLI::AbortSilent
+        end
```

Comment:
How come these changes were made? If this form returns nil, a new script is created on the server during a push. It currently is not possible to create a script any other way, so I don't think we want to abort in these cases

Reply:
I'll revert the changes because both approaches [lead to return `nil`](https://github.com/Shopify/shopify-cli/blob/main/lib/shopify_cli/form.rb#L15). Regarding the creation of the script on push, would you mind pointing me to the place in the codebase where the script creation takes place?

Reply:
👍 Thanks! The CLI code makes a `set_app_script` GraphQL mutation call [here](https://github.com/Shopify/shopify-cli/blob/8a2f7ae276503ebf8f447f3045c24d8ce5398ede/lib/project_types/script/layers/infrastructure/script_service.rb#L15-L26). The uuid in the mutation is nilable, and ScriptService [checks its presence](https://github.com/Shopify/script-service/blob/3d42fcfe663cfc3454059f3d414ea4c121e3a8c8/lib/core_integration/application/set_app_script.rb#L69-L98) to figure out if it should create a script or now